tests/TestCase/Controller/ControllerFactoryTest.php

```diff
@@ -757,9 +755,22 @@ public function testInvokePassedParametersCoercion(): void
 
         $result = $this->factory->invoke($controller);
         $data = json_decode((string)$result->getBody(), true);
-
-        $this->assertNotNull($data);
         $this->assertSame(['one' => 1.0, 'two' => 2, 'three' => false, 'four' => []], $data);
+
+        $request = new ServerRequest([
+            'url' => 'test_plugin_three/dependencies/requiredTyped',
+            'params' => [
+                'plugin' => null,
+                'controller' => 'Dependencies',
+                'action' => 'requiredTyped',
+                'pass' => ['1.0', '-1', '0', ''],
```

Comment:
```suggestion
                'pass' => ['1.0', '-01', '0', ''],
```

Reply:
`-01` shouldn't be coerced to int IMO.

Reply:
Do we break it now?

Reply:
Personally I would prefer a strict behavior but breaking a tested behavior isn't nice in general.

Reply:
Think the reason we supported leading 0's was partly just because we used ctype_digit() and partly because I've seen leading 0's for IDs in URLs and probably thought that was valid.

I don't know if I didn't support negative numbers on purpose, or just happened that way.